### PR TITLE
Rest of data generators

### PR DESCRIPTION
**Pull request by jiayu-google**
1. Add heterogenous and heavy tailed impression generator.
2. Add overlap set generators.
3. Add a line in requirements.txt to allow easy import of modules in cardinality_estimation_evaluation_framework.

### CODE REVIEW

**Review by uakyol — APPROVED** [review]

**Review by matthewclegg — CHANGES_REQUESTED** [review]

File: src/data_generators/heavy_tailed_impression_generator.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+class HeavyTailedImpressionGenerator(ImpressionGenerator):
+  """Generate impressions with heavy tailed impressions."""
+
+  def __init__(self, n: int, zipf_s: float = 2,
+               random_state: RandomState = None):
+    """Constructor for the HeavyTailedImpressionGenerator.
+
+    For each user, the number of impressions assigned to that user is
+    determined by drawing from a zipf distribution which has pmf
```
[the remark this answers is not in the document]
Comment (jiayu-google) [reply src/data_generators/heavy_tailed_impression_generator.py:32]: Changed every zipf to zeta, except in np.random.zipf.
In fact np.random.zipf generates a zeta distr, and there is no np.random.zeta...

File: src/data_generators/heterogeneous_impression_generator.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    """Constructor for the HeterogeneousImpressionGenerator.
+
+    For each user, the number of impressions assigned to that user is
+    determined by drawing from a shifted (by +1) Poisson distribution with
+    parameter lambda, with lambda drawn from a Gamma distribution. Explicitly,
+    lambda is generated from a pdf
+    p(x) = x^{shape - 1} * exp(x / scale) / scale^shape / Gamma(shape),
+    where shape and gamma are two parameters and Gamma is the Gamma function.
+    In this way, lambda has a mean of shape * scale and a variance of
+    shape * scale^2.
```
[the remark this answers is not in the document]
Comment (jiayu-google) [reply src/data_generators/heterogeneous_impression_generator.py:38]: Good point!

File: src/data_generators/overlap_data_set.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+from wfa_planning_evaluation_framework.data_generators.data_set import (
+    DataSet,
+)
+from wfa_planning_evaluation_framework.data_generators.publisher_data import (
+    PublisherData,
+)
+
+
+class OverlapDataSet(DataSet):
+  """Construct a multi-pub DataSet with from list of single-pub PublisherData.
```
[the remark this answers is not in the document]
Comment (jiayu-google) [reply src/data_generators/overlap_data_set.py:33]: Done

**Review by matthewclegg — CHANGES_REQUESTED** [review]
Please reformat all files using 'black'.  Otherwise, looks good!

**Review by matthewclegg — APPROVED** [review]

File: src/simulator/tests/publisher_test.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def test_true_reach_by_spend(self):
 
     def test_liquid_legions_sketch(self):
         sketch = self.publisher.liquid_legions_sketch(0.04)
-        estimator = StandardizedHistogramEstimator(epsilon=np.Inf)
+        estimator = StandardizedHistogramEstimator()
```
Comment (my2457) [inline src/simulator/tests/publisher_test.py:53]: same as in #16 , getting test failures here with the change

Comment (matthewclegg) [reply src/simulator/tests/publisher_test.py:53]: See previous comment.

**Pull request merged**